Makefile

```diff
@@ -62,3 +62,169 @@ release:
+
+## Location to install dependencies to
+
+# Use CACHE_BIN for tools that cannot use devbox and LOCALBIN for tools that can use either method
+CACHE_BIN ?= $(CURDIR)/bin
+LOCALBIN ?= $(CACHE_BIN)
+
+DEVBOX_BIN ?= $(DEVBOX_PACKAGES_DIR)/bin
+
+# if the $DEVBOX_PACKAGES_DIR env variable exists that means we are within a devbox shell and can safely
+# use devbox's bin for our tools
+ifdef DEVBOX_PACKAGES_DIR
+	LOCALBIN = $(DEVBOX_BIN)
+endif
+
+export PATH := $(CACHE_BIN):$(PATH)
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+##@ Tooling Binaries:
+CTLPTL         ?= $(LOCALBIN)/ctlptl
+CLUSTERCTL     ?= $(LOCALBIN)/clusterctl
+KIND           ?= $(LOCALBIN)/kind
+KUSTOMIZE      ?= $(LOCALBIN)/kustomize
+CHAINSAW      ?= $(LOCALBIN)/chainsaw
+KUBECTL        ?= $(LOCALBIN)/kubectl
+YQ            ?= $(LOCALBIN)/yq
+ENVSUBST      ?= $(LOCALBIN)/envsubst
+
+## Tool Versions
+CTLPTL_VERSION           ?= v0.8.28
+CLUSTERCTL_VERSION       ?= v1.6.3
+KIND_VERSION             ?= v0.22.0
+KUSTOMIZE_VERSION        ?= v5.3.0
+CHAINSAW_VERSION         ?= v0.2.2
+
+.PHONY: tools
+tools: ctlptl clusterctl kind kustomize chainsaw kubectl yq envsubst
+
+.PHONY: ctlptl
+ctlptl: $(CTLPTL) ## Download ctlptl locally if necessary.
+$(CTLPTL): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) go install github.com/tilt-dev/ctlptl/cmd/ctlptl@$(CTLPTL_VERSION)
+
+.PHONY: clusterctl
+clusterctl: $(CLUSTERCTL) ## Download clusterctl locally if necessary.
+$(CLUSTERCTL): $(LOCALBIN)
+	curl -fsSL https://github.com/kubernetes-sigs/cluster-api/releases/download/$(CLUSTERCTL_VERSION)/clusterctl-$(OS)-$(ARCH_SHORT) -o $(CLUSTERCTL)
+	chmod +x $(CLUSTERCTL)
+
+.PHONY: kind
+kind: $(KIND) ## Download kind locally if necessary.
+$(KIND): $(LOCALBIN)
+	curl -fsSL https://github.com/kubernetes-sigs/kind/releases/download/$(KIND_VERSION)/kind-$(OS)-$(ARCH_SHORT) -o $(KIND)
+	chmod +x $(KIND)
+
+.PHONY: kubectl
+kubectl: $(KUBECTL) ## Download kubectl locally if necessary.
+$(KUBECTL): $(LOCALBIN)
+	curl -L https://dl.k8s.io/release/$(shell curl -L -s https://dl.k8s.io/release/stable.txt)/bin/$(OS)/$(ARCH_SHORT)/kubectl -o $(KUBECTL)
+	chmod +x $(KUBECTL)
+
+.PHONY: yq
+yq: $(YQ) ## Download yq locally if necessary.
+$(YQ): $(LOCALBIN)
+	curl -L https://github.com/mikefarah/yq/releases/latest/download/yq_$(OS)_$(ARCH_SHORT) -o $(YQ)
+	chmod +x $(YQ)
+
+.PHONY: envsubst
+envsubst: $(ENVSUBST) ## Download envsubst locally if necessary.
+$(ENVSUBST): $(LOCALBIN)
+	curl -L https://github.com/a8m/envsubst/releases/download/v1.2.0/envsubst-$(OS_UPPER)-$(ARCH) -o $(ENVSUBST)
+	chmod +x $(ENVSUBST)
+
+.PHONY: kustomize
+kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
+$(KUSTOMIZE): $(LOCALBIN)
+	GOBIN=$(LOCALBIN) GO111MODULE=on go install sigs.k8s.io/kustomize/kustomize/v5@$(KUSTOMIZE_VERSION)
+
+.PHONY: chainsaw
+chainsaw: $(CHAINSAW) ## Download chainsaw locally if necessary.
+$(CHAINSAW): $(CACHE_BIN)
+	GOBIN=$(CACHE_BIN) go install github.com/kyverno/chainsaw@$(CHAINSAW_VERSION)
```

Comment:
Are all of these tools available through devbox? If they are, could we avoid this type of tool management, and rely on devbox to make sure everything is available? Devbox even has [an action](https://www.jetify.com/devbox/docs/continuous_integration/github_action/) to set up the environment.

Reply:
Oh there is an action for that? I'll just use devbox then!